splunk_add_on_ucc_framework/auto_gen_comparator.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright 2024 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from os import walk
+from os.path import sep
+from typing import List, Dict
+from logging import Logger
+
+
+class CodeGeneratorDiffChecker:
+    def __init__(self, src_dir: str, dst_dir: str) -> None:
+        self.source_directory = src_dir
+        self.target_directory = dst_dir
+        self.common_files: Dict[str, str] = {}
+
+    def find_common_files(
+        self, logger: Logger, ignore_file_list: List[str] = []
+    ) -> None:
+        # we add these two files as they are required to be present in source code
+        ignore_file_list.extend(["app.manifest", "README.txt"])
+
+        src_all_files = {}
+        for root, _, files in walk(self.source_directory):
+            for file in files:
+                src_all_files[file] = sep.join([root, file])
+
+        dest_all_files = []
+        for root, _, files in walk(self.target_directory):
+            for file in files:
+                dest_all_files.append(file)
+        dest_all_files.append("default.meta")
+
+        for file_name in dest_all_files:
+            if file_name in src_all_files.keys():
+                if file_name in ignore_file_list:
```

Comment:
`ignore_file_list` could be a set, because when it is a list, there would be a lot of comparisons when `ignore_file_list` is big. But I assume that it will rather be small so I think it is ok.